Assets/Plugins/Editor/Uplift/Schemas/Repositories/GithubRepository.cs

```diff
@@ -74,35 +74,50 @@ public override Upset[] ListPackages()
 
             List<Upset> upsetList = new List<Upset>();
             EditorUtility.DisplayProgressBar(progressBarTitle, "Please wait a little bit while Uplift parses the Upset in the GitHub repository at " + urlField, 0f);
+
             string assetPath;
-            foreach (GitHubAsset asset in upsetAssets)
+            try
             {
-                StrictXmlDeserializer<Upset> deserializer = new StrictXmlDeserializer<Upset>();
-                
-                EditorUtility.DisplayProgressBar(
-                    progressBarTitle,
-                    "Parsing " + asset.name, 
-                    (float)(index++) / upsetAssets.Length
-                );
-
-                if (!TryGetCachedItem(asset.name, out assetPath))
+                foreach (GitHubAsset asset in upsetAssets)
                 {
-                    using (StreamReader sr = new StreamReader(GitHub.GetAssetStream(asset, GetToken())))
-                    using (FileStream fs = new FileStream(assetPath, FileMode.Create))
+                    try
                     {
-                        sr.BaseStream.CopyTo(fs);
+                        StrictXmlDeserializer<Upset> deserializer = new StrictXmlDeserializer<Upset>();
+
+                        EditorUtility.DisplayProgressBar(
+                            progressBarTitle,
+                            "Parsing " + asset.name,
+                            (float)(index++) / upsetAssets.Length
+                        );
+
+                        if (!TryGetCachedItem(asset.name, out assetPath))
+                        {
+                            using (StreamReader sr = new StreamReader(GitHub.GetAssetStream(asset, GetToken())))
+                            using (FileStream fs = new FileStream(assetPath, FileMode.Create))
+                            {
+                                sr.BaseStream.CopyTo(fs);
+                            }
+                        }
+
+                        using (FileStream fs = new FileStream(assetPath, FileMode.Open))
+                        {
+                            Upset upset = deserializer.Deserialize(fs);
+                            upset.MetaInformation.dirName = asset.name;
+                            upsetList.Add(upset);
+                        }
+                    }
+                    catch(Exception e)
+                    {
+                        Debug.LogErrorFormat("An error occured while trying to get asset {0} ({1}, {2}) from GithubRepository", asset.name, asset.htmlURL, asset.apiURL);
```

Comment:
don't we lose the error here? Maybe LogException as well in case it isn't what we think it is.

Reply:
I'm not sure why we would lose the error? It would still be logged